File: pipeline/zReferenceFiles/ReferenceFilesForCodesofBase64,DefaultandSolutions/NodeJS/NonNodeTypeSolutionCode.js
class Solution {
    static findMaxMinRuntime(numAPIs, runtimes, numApps) {
        let low = 0;
        let high = runtimes.reduce((sum, r) => sum + r, 0);
        
        let ans = 0;
        while (low <= high) {
            let mid = Math.floor((low + high) / 2);
            if (mid === 0) {
                low = 1;
                continue;
            }
            
            let totalProvided = runtimes.reduce((sum, r) => sum + Math.min(r, mid), 0);
            
            if (totalProvided >= mid * numApps) {
                ans = mid;
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return ans;
    }
}

File: pipeline/zReferenceFiles/ReferenceFilesForCodesofBase64,DefaultandSolutions/Cpp/NonNodeTypeSolutionCode.cpp
#include <bits/stdc++.h>
using namespace std;

class solution {
public:
    long long findMaxMinRuntime(int numAPIs, vector<int>& runtimes, int numApps) {
        long long low = 0;
        long long high = 0;
        for (int x : runtimes) high += x;
        high /= numApps;
        long long ans = 0;
        
        high = 2e14;
        auto check = [&](long long mid) {
            long long total_provided = 0;
            for (int r : runtimes) {
                total_provided += min((long long)r, mid);
            }
            return total_provided >= (mid * numApps);
        };

        while (low <= high) {
            long long mid = low + (high - low) / 2;
            if (mid == 0) {
                low = 1;
                continue;
            }
            if (check(mid)) {
                ans = mid;
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return ans;
    }
};
